File: Python/fib.py
def fib(n, lookup):
	if n <= 1:
		return n
	
	if n not in lookup:
		lookup[n] = fib(n-1, lookup) + fib(n-2, lookup)
	print(lookup)
	return lookup[n]

if __name__ == '__main__':
	n = 9
	lookup = {}
	print(fib(n, lookup))

File: Java/Fib.java
public class Fib
{
	static int Fibonacci(int n)
	{
		if(n <= 1)
			return n;
		
		return Fibonacci(n-1) + Fibonacci(n-2);
	}
	
	public static void main(String[] args)
	{
		for(int i = 1; i < 15; i++)
		{
			System.out.print(Fibonacci(i) + "  ");
		}
		System.out.println();
	}
}
